simulation/__main__.py

```diff
@@ -199,10 +195,22 @@ def validate_args(args):
             f"Boiler ban announcement date must be on or before ban date, got gas_oil_boiler_ban_date:{args.gas_oil_boiler_ban_date}, gas_oil_boiler_ban_announce_date:{args.gas_oil_boiler_ban_announce_date}"
         )
 
-    if args.campaign_target_heat_pump_awareness < args.heat_pump_awareness:
-        raise ValueError(
-            f"Campaign target awareness must be greater than or equal to the population heat pump awareness, got campaign_target_heat_pump_awareness:{args.campaign_target_heat_pump_awareness}, heat_pump_awareness:{args.heat_pump_awareness}"
+    if args.campaign_target_heat_pump_awareness_date is not None:
+        # Check that target awareness inputs increase over the model horizon
+        campaigns = sorted(args.campaign_target_heat_pump_awareness_date)
```

Comment:
This should be a function that we unit test I think!

Reply:
See `test_campaign_target_less_than_heat_pump_awareness_raises_value_error` in `test_main`

Reply:
Oh I mean the logic inside this `if` statement - it's a pretty convoluted logic that I think we should write as a function, then test that given inputs you get expected output.

e.g.

```python
     if args.campaign_target_heat_pump_awareness_date is not None:
           x = parse_campaign_dates_and_target(arg1, arg2)

     def test_parse_campaign_dates_and_target():
           test this function.
```

Increase legibility of the code, and if someone changes some stuff then this test will break, and they'd have to either update the tests (if break is intended) or fix the change to conform to the test etc.